src/descriptor.rs

```diff
@@ -83,6 +83,242 @@ pub struct KeyboardReport {
     pub keycodes: [u8; 6],
 }
 
+/// KeyboardUsage describes the key codes to be used in implementing a USB keyboard.
+///
+/// The usage type of all key codes is Selectors, except for the modifier keys
+/// Keyboard Left Control to Keyboard Right GUI which are Dynamic Flags.
+///
+/// Reference: <https://usb.org/sites/default/files/hut1_3_0.pdf>
+///
+#[repr(u8)]
+#[allow(unused)]
+#[non_exhaustive]
+#[derive(Copy, Debug, Clone, Eq, PartialEq)]
+pub enum KeyboardUsage {
+    // 0x00: Reserved
+    KeyboardErrorRollOver = 0x01,
+    KeyboardPOSTFail = 0x02,
+    KeyboardErrorUndefined = 0x03,
+    KeyboardAa = 0x04,
+    KeyboardBb = 0x05,
+    KeyboardCc = 0x06,
+    KeyboardDd = 0x07,
+    KeyboardEe = 0x08,
+    KeyboardFf = 0x09,
+    KeyboardGg = 0x0A,
+    KeyboardHh = 0x0B,
+    KeyboardIi = 0x0C,
+    KeyboardJj = 0x0D,
+    KeyboardKk = 0x0E,
+    KeyboardLl = 0x0F,
+    KeyboardMm = 0x10,
+    KeyboardNn = 0x11,
+    KeyboardOo = 0x12,
+    KeyboardPp = 0x13,
+    KeyboardQq = 0x14,
+    KeyboardRr = 0x15,
+    KeyboardSs = 0x16,
+    KeyboardTt = 0x17,
+    KeyboardUu = 0x18,
+    KeyboardVv = 0x19,
+    KeyboardWw = 0x1A,
+    KeyboardXx = 0x1B,
+    KeyboardYy = 0x1C,
+    KeyboardZz = 0x1D,
+    Keyboard1Bang = 0x1E,
+    Keyboard2At = 0x1F,
+    Keyboard3Hash = 0x20,
+    Keyboard4Dollar = 0x21,
+    Keyboard5Percent = 0x22,
+    Keyboard6Caret = 0x23,
+    Keyboard7Ampersand = 0x24,
+    Keyboard8Star = 0x25,
```

Comment:
`*` should be Asterisk https://en.wikipedia.org/wiki/List_of_typographical_symbols_and_punctuation_marks